src/KronFluxAlgorithm.cc

```diff
@@ -525,40 +488,41 @@ void KronFlux::_applyAperture(
             }
         }
         double const moment = irSum/iSum;
-        R_K_psf = ::sqrt(afw::geom::PI/2)*::hypot(moment, std::max(0.0, ctrl.smoothingSigma));
-        if (ctrl.enforceMinimumRadius && rad < R_K_psf) {
+        R_K_psf = ::sqrt(afw::geom::PI/2)*::hypot(moment, std::max(0.0, _ctrl.smoothingSigma));
+        if (_ctrl.enforceMinimumRadius && rad < R_K_psf) {
             aperture.getAxes().scale(R_K_psf/rad);
-            source.set(_smallRadiusKey, true); // guilty after all
+            _flagHandler.setValue(source, SMALL_RADIUS, true);
         }
     }
 
-    std::pair<double, double> result = aperture.measure(exposure.getMaskedImage(), ctrl.nRadiusForFlux,
-                                                        ctrl.maxSincRadius);
-    source.set(getKeys().meas, result.first);
-    source.set(getKeys().err, result.second);
+    std::pair<double, double> result = aperture.measure(exposure.getMaskedImage(), _ctrl.nRadiusForFlux,
+                                                        _ctrl.maxSincRadius);
+    meas::base::FluxResult fluxResult;
+    fluxResult.flux = result.first;
+    fluxResult.fluxSigma = result.second;
+    source.set(_fluxResultKey, fluxResult);
     source.set(_radiusKey, aperture.getAxes().getDeterminantRadius());
     //
     // Now aperture corrections. Calculate the PSF models' Kron flux, and allow
     // the aperture correction code to force Kron fluxes to agree with the PSF flux for point sources
     //
     double const psfFactor = getPsfFactor(exposure.getPsf(), aperture.getCenter(),
-                                          R_K_psf*ctrl.nRadiusForFlux, ctrl.maxSincRadius);
-    source.set(_fluxCorrectionKeys.psfFactor, psfFactor);
-    source.set(_fluxCorrectionKeys.psfFactorFlag, false); // i.e. good
+                                          R_K_psf*_ctrl.nRadiusForFlux, _ctrl.maxSincRadius);
+
+    //  FIXME:  This need to be done after fluxCorrection is implemented in meas_base
+    //source.set(_fluxCorrectionKeys.psfFactor, psfFactor);
+    //source.set(_fluxCorrectionKeys.psfFactorFlag, false); // i.e. good
```

Comment:
I don't like leaving commented, "FIXME" code in the source -- we can't rely on folks remembering or happening to notice it in future. If we know work has to be done in the future, it should be filed as an issue in JIRA and the comment should refer to that issue.